GVFS/GVFS.UnitTests/Tracing/QueuedPipeStringWriterTests.cs

```diff
@@ -0,0 +1,265 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Pipes;
+using System.Linq;
+using System.Threading;
+using GVFS.Common.Tracing;
+using NUnit.Framework;
+
+namespace GVFS.UnitTests.Tracing
+{
+    [TestFixture]
+    public class QueuedPipeStringWriterTests
+    {
+        [TestCase]
+        public void Stop_RaisesStateStopped()
+        {
+            QueuedPipeStringWriter writer = new QueuedPipeStringWriter(
+                () => null); // Not important since the test will never enqueue any messaged to write
+
+            // Capture event invocations
+            List<QueuedPipeStringWriter.QueuedPipeStringWriterStateEventArgs> eventCalls = new List<QueuedPipeStringWriter.QueuedPipeStringWriterStateEventArgs>();
+            writer.StateChanged += (_, e) => eventCalls.Add(e);
+
+            // Try to write some dummy data
+            writer.Start();
+            writer.Stop();
+
+            Assert.AreEqual(1, eventCalls.Count);
+            Assert.AreEqual(QueuedPipeStringWriter.QueuedPipeStringWriterState.Stopped, eventCalls[0].State);
```

Comment:
Comment applies to all the `Assert` checks in the file:

Switch from using `Assert` syntax to using the `Should` syntax that's used elsewhere in the unit tests (you'll need to add a `using GVFS.Tests.Should`)